cypress/e2e/integration/features/load-simulation-from-sid-parameter.cy.js

```diff
@@ -0,0 +1,58 @@
+import {
+  SAVE_MODAL_EMAIL_INPUT,
+  SAVE_MODAL_SUBMIT_BUTTON,
+} from '../../../constants/elements-ids'
+import { TEST_USER_ID } from '../../../constants/user'
+import {
+  clickNextButton,
+  clickSkipTutorialButton,
+} from '../../../helpers/elements/buttons'
+import { visit } from '../../../helpers/interactions/visit'
+
+describe('Loading the simulation from the sid parameter', () => {
+  context(
+    'given the user saves their simulation after answering a question',
+    () => {
+      let simulationId
+
+      beforeEach(() => {
+        cy.intercept('POST', '/simulations/create', (req) => {
+          req.body.userId = TEST_USER_ID
+        }).as('saveSimulation')
```

Comment:
No way to modify localStorage ? This boilerplate needs to be replicated.

Reply:
No it didn't seem to work, I couldn't find why. I think that perhaps even if the localStorage was modified, the internal `userId` value stored in publicodes-state doesn't get refreshed upon change in the localStorage. The timing to set the localStorage `userId` right before it is loaded by publicodes-state is short.

> This boilerplate needs to be replicated.
 I'm wondering if it's a good idea, the `as('saveSimulation')` would still need to be visible in order for the dev to use the right label (saveSimulation) below ; also the route will change for each new route and the method might also. With that in mind, what would you suggest?